sw/device/lib/base/status.h

```diff
@@ -104,6 +104,44 @@ typedef struct status {
     status_.value;                         \
   })
 
+/**
+ * Evaluates a status_t for Ok or Error status, returning the Ok value.
+ *
+ * This macro is like the `try!` macro (or now `?` operator) in Rust:
+ * It evaluates to the contained OK value or it immediately returns from
+ * the enclosing function with the error value.
+ *
+ * @param expr_ An expression that can be converted to a `status_t`.
+ * @return The enclosed OK value.
+ */
+#define ASSERT_TRUE(expr_, ...)     \
+  ({                                \
+    if (CHECK_IMPL(expr_)) {        \
```

Comment:
Should there be an explicit `#include "sw/device/lib/testing/test_framework/check.h"` for the `CHECK_IMPL` and `CHECK_ARRAYS_EQ_IMPL` macros?

Reply:
I don't think `status.h` should depend on `check.h` because one is in the base (using for the ROM for example)  while the other is only for testing. I think either we should move those macros to `check.h` or have the generic part of those checks in `status.h` or another file in `lib/base/`